### PR TITLE
[bitnami/kubernetes-event-exporter] Introduce kubernetes-event-exporter

### PR DESCRIPTION
**Pull request by mkilchhofer**
**Description of the change**
Contribute a new helm chart for [opsgenie/kubernetes-event-exporter](https://github.com/opsgenie/kubernetes-event-exporter/)

**Benefits**
Deploy the [opsgenie/kubernetes-event-exporter](https://github.com/opsgenie/kubernetes-event-exporter/) via a Helm Chart

**Possible drawbacks**
`-`

**Applicable issues**
`-`

**Additional information**
More infos about the software can be found here: https://github.com/opsgenie/kubernetes-event-exporter/

**Checklist** 
- [x] Discuss / organize bitnami container image build process
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by mkilchhofer** [comment]
If you wish to accept this contribution, we need to discuss about https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md
> The chart should use Bitnami based container images. If they don't exist, you can open a GitHub issue and we will work together to create them.

**Comment by FraPazGal** [comment]
Hi @mkilchhofer,

Thank you so much for this PR!

I'll start creating the bitnami image of kubernetes-event-exporter for this chart and adding it to our catalog. Meanwhile, I'll review this PR for it to be ready to merge to our repo once the bitnami image is done.

**Comment by mkilchhofer** [comment]
Not stale

**Comment by javsalgar** [comment]
Hi,

Just a note to let you know that we released the kubernetes-event-exporter container: https://hub.docker.com/repository/docker/bitnami/kubernetes-event-exporter/tags?page=1

Could you change the PR so it now uses this image? Could you check that there are no issues with the chart when changing the image?

**Comment by mkilchhofer** [comment]
Okay, I updated this PR and it works. Is it true that the container build repository (containing Dockerfiles etc.) isn't yet published?
What I observe is that the Debug Log output is a bit different regarding the golang package names (I don't know how they are called correctly):

Official:
```log
2020-10-23T11:14:51Z INF app/pkg/exporter/engine.go:25 > Registering sink name=dump type=*sinks.File
2020-10-23T11:14:51Z DBG app/pkg/kube/watcher.go:63 > Received event involvedObject=kubernetes-event-exporter msg="Scaled up replica set kubernetes-event-exporter-64b4f4bbd to 1" namespace=monitoring reason=ScalingReplicaSet
2020-10-23T11:14:51Z DBG app/pkg/kube/watcher.go:63 > Received event involvedObject=kubernetes-event-exporter-64b4f4bbd msg="Created pod: kubernetes-event-exporter-64b4f4bbd-2xctb" namespace=monitoring reason=SuccessfulCreate
2020-10-23T11:14:51Z DBG app/pkg/exporter/channel_registry.go:56 > sending event to sink event="Scaled up replica set kubernetes-event-exporter-64b4f4bbd to 1" sink=dump
{"component":"deployment-controller","count":"2","host":"","kind":"Deployment","message":"Scaled up replica set kubernetes-event-exporter-64b4f4bbd to 1","name":"kubernetes-event-exporter","namespace":"monitoring","reason":"ScalingReplicaSet","type":"Normal"}
2020-10-23T11:14:52Z DBG app/pkg/kube/watcher.go:63 > Received event involvedObject=kubernetes-event-exporter-7ffdbb4bdd msg="Deleted pod: kubernetes-event-exporter-7ffdbb4bdd-lct5n" namespace=monitoring reason=SuccessfulDelete
2020-10-23T11:14:52Z DBG app/pkg/exporter/channel_registry.go:56 > sending event to sink event="Created pod: kubernetes-event-exporter-64b4f4bbd-2xctb" sink=dump
```

Your build:
```log
2020-10-23T11:11:17Z INF bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/exporter/engine.go:25 > Registering sink name=dump type=*sinks.File
2020-10-23T11:11:17Z DBG bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/watcher.go:63 > Received event involvedObject=kubernetes-event-exporter-7556478469-bvw5s msg="Successfully pulled image \"docker.io/bitnami/kubernetes-event-exporter:0.9.0-scratch-r1\"" namespace=monitoring reason=Pulled
2020-10-23T11:11:18Z DBG bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/watcher.go:63 > Received event involvedObject=kubernetes-event-exporter-7556478469-bvw5s msg="Created container event-exporter" namespace=monitoring reason=Created
2020-10-23T11:11:18Z DBG bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/exporter/channel_registry.go:56 > sending event to sink event="Created container event-exporter" sink=dump
2020-10-23T11:11:18Z DBG bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/watcher.go:63 > Received event involvedObject=kubernetes-event-exporter-7556478469-bvw5s msg="Started container event-exporter" namespace=monitoring reason=Started
2020-10-23T11:11:18Z DBG bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/kube/watcher.go:63 > Received event involvedObject=kubernetes-event-exporter msg="Scaled down replica set kubernetes-event-exporter-7ffdbb4bdd to 0" namespace=monitoring reason=ScalingReplicaSet
{"component":"kubelet","count":"1","host":"k3s-node3","kind":"Pod","message":"Created container event-exporter","name":"kubernetes-event-exporter-7556478469-bvw5s","namespace":"monitoring","reason":"Created","type":"Normal"}
2020-10-23T11:11:18Z DBG bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/exporter/channel_registry.go:56 > sending event to sink event="Successfully pulled image \"docker.io/bitnami/kubernetes-event-exporter:0.9.0-scratch-r1\"" sink=dump
{"component":"kubelet","count":"1","host":"k3s-node3","kind":"Pod","message":"Successfully pulled image \"docker.io/bitnami/kubernetes-event-exporter:0.9.0-scratch-r1\"","name":"kubernetes-event-exporter-7556478469-bvw5s","namespace":"monitoring","reason":"Pulled","type":"Normal"}
```

Do you see the difference right after the log level?
```log
app/pkg/exporter/engine.go:25
vs.
bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/exporter/engine.go:25
```
Is that an desired change from your side?

**Comment by javsalgar** [comment]
Hi,

The container dockerfile is here: https://github.com/bitnami/bitnami-docker-kubernetes-event-exporter

**Comment by javsalgar** [comment]
Could it be because the Go Version used in each of them? Maybe they are different

**Comment by mkilchhofer** [comment]
> Hi,
> 
> The container dockerfile is here: https://github.com/bitnami/bitnami-docker-kubernetes-event-exporter

I get a 404 error. Is it a private repo? I am not part of you guys :)

**Comment by javsalgar** [comment]
Damn! It was incorrectly set as private. Try again

**Comment by mkilchhofer** [comment]
Okay I can now access :)
But I cannot find any sourcecode of the binary file here: https://github.com/bitnami/bitnami-docker-kubernetes-event-exporter/blob/master/0/scratch/rootfs/kubernetes-event-exporter

When I checkout this repo and run the binary directly, logging is already noisy:
```console
$ pwd
/tmp/bitnami-docker-kubernetes-event-exporter/0/scratch/rootfs
$ ls
config.yaml  kubernetes-event-exporter
$ cat config.yaml 
logLevel: debug
logFormat: pretty
receivers:
  - name: "dump"
    file:
      path: "/dev/stdout"
route:
  routes:
    - match:
        - receiver: "dump"
$ ./kubernetes-event-exporter 
2020-10-24T10:41:53+02:00 INF bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/exporter/engine.go:25 > Registering sink name=dump type=*sinks.File
^C2020-10-24T10:41:58+02:00 INF bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/main.go:103 > Received signal to exit signal=interrupt
2020-10-24T10:41:58+02:00 INF bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/exporter/engine.go:43 > Closing sinks
2020-10-24T10:41:58+02:00 INF bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/exporter/channel_registry.go:62 > Closing the sink sink=dump
2020-10-24T10:41:58+02:00 INF bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/exporter/channel_registry.go:67 > Closed sink=dump
2020-10-24T10:41:58+02:00 INF bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/pkg/exporter/engine.go:45 > All sinks closed
2020-10-24T10:41:58+02:00 INF bitnami/blacksmith-sandox/kubernetes-event-exporter-0.9.0/src/github.com/opsgenie/kubernetes-event-exporter/main.go:98 > Exiting
```

File: bitnami/kubernetes-event-exporter/README.md
Comment (FraPazGal) [inline bitnami/kubernetes-event-exporter/README.md:14]:
```suggestion
This chart bootstraps a [Kubernetes Event Exporter](https://github.com/bitnami/bitnami-docker-kubernetes-event-exporter) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
```

File: bitnami/kubernetes-event-exporter/README.md
Comment (FraPazGal) [inline bitnami/kubernetes-event-exporter/README.md:47]:
```suggestion
The following table lists the configurable parameters of the Kubernetes Event Exporter chart and their default values.
```

Small typo.

File: bitnami/kubernetes-event-exporter/README.md
Comment (FraPazGal) [inline bitnami/kubernetes-event-exporter/README.md:49]: Could you please divide this table in several smaller ones? It would help readability. You can use one of our [other READMEs ](https://github.com/bitnami/charts/tree/master/bitnami/opencart#parameters) as reference. 

Also, there are several missing parameters that are already included in `values.yaml` like `global.xxx` or `nameOverride`

File: bitnami/kubernetes-event-exporter/templates/deployment.yaml
Comment (FraPazGal) [inline bitnami/kubernetes-event-exporter/templates/deployment.yaml:36]:
```suggestion
      initContainers: {{- include "common.tplvalues.render" ( dict "value" .Values.initContainers "context" $ ) | nindent 6 }}
```

I think this can be in a single line.

File: bitnami/kubernetes-event-exporter/templates/deployment.yaml
Comment (FraPazGal) [inline bitnami/kubernetes-event-exporter/templates/deployment.yaml:45]:
```suggestion
          {{- if .Values.containerSecurityContext.enabled }}
          securityContext: {{- omit .Values.containerSecurityContext "enabled" | toYaml | nindent 12 }}
          {{- end }}
```

Just to conform to the new standard we are in progress of implementing in our charts, it would be good to change the parameter's name and add the `.enabled` option. What do you think?

File: bitnami/kubernetes-event-exporter/values-production.yaml
Comment (FraPazGal) [inline bitnami/kubernetes-event-exporter/values-production.yaml:86]: Could you please change it to the user `1001`? It is the default non-root one used in all our containers/charts.

File: bitnami/kubernetes-event-exporter/values-production.yaml
Comment (FraPazGal) [inline bitnami/kubernetes-event-exporter/values-production.yaml:17]: I think that we could include the `commonAnnotations` and `commonLabels` parameters to allow for custom annotation/labels across all deployed resources, what do you think?

**Review by FraPazGal — COMMENTED** [review]
Hello @mkilchhofer!

Great work with this PR! Just a few comments, most of them style-related. Could you please give them a look?

Once the changes are done, the remaining step would be for the team to create the chart tests for our CI/CD tools and get the green light.

Also, you can delete the `values-production.yaml` file, as it is something that is in our roadmap to be removed from all of our charts.

Regarding the difference in the packages name, I would say that is the expected behaviour and it stems from our compilation tool. I agree it is a non-blocker for this PR.

File: bitnami/kubernetes-event-exporter/templates/deployment.yaml
Comment (mkilchhofer) [reply bitnami/kubernetes-event-exporter/templates/deployment.yaml:44]: Oh yes, I didn't notice this refactoring yet.

File: bitnami/kubernetes-event-exporter/values-production.yaml
Comment (mkilchhofer) [reply bitnami/kubernetes-event-exporter/values-production.yaml:17]: Sure, I implemented both.

File: bitnami/kubernetes-event-exporter/values-production.yaml
Comment (mkilchhofer) [reply bitnami/kubernetes-event-exporter/values-production.yaml]: Done

File: bitnami/kubernetes-event-exporter/README.md
Comment (mkilchhofer) [reply bitnami/kubernetes-event-exporter/README.md]: I updated the params table and divided them into smaller sections. Is it okay like this?

Comment (FraPazGal) [reply bitnami/kubernetes-event-exporter/README.md]: Looks great! It seems only the `image.pullSecrets` parameter is still missing on the README.

File: bitnami/kubernetes-event-exporter/Chart.yaml
Comment (FraPazGal) [inline bitnami/kubernetes-event-exporter/Chart.yaml:5]:
```suggestion
appVersion: 0.9.0
```
Just a minor change, as our internal tools expects `appVersion` to follow semver standards.

Comment (mkilchhofer) [reply bitnami/kubernetes-event-exporter/Chart.yaml:5]: I agree that everyone should use SemVer for versioning their stuff. But the upstream version of this event exporter only uses ‘0.9‘. IMHO we should use it like this. But when your tests are failing because of this, I agree for going the easy way :-)

File: bitnami/kubernetes-event-exporter/README.md
Comment (mkilchhofer) [reply bitnami/kubernetes-event-exporter/README.md]: I am currently in the mountains for a few days. I will fix it next monday 😃

File: bitnami/kubernetes-event-exporter/Chart.yaml
Comment (FraPazGal) [reply bitnami/kubernetes-event-exporter/Chart.yaml:5]: Yes, this was something that also bothered me. I looked into it and the error comes from our tool expecting a string but receiving a float. As such, I think that this can be solved by just changing it to `appVersion: v0.9`.

File: bitnami/kubernetes-event-exporter/README.md
Comment (mkilchhofer) [reply bitnami/kubernetes-event-exporter/README.md]: Done :)

**Comment by FraPazGal** [comment]
Hello @mkilchhofer!

Just to give you an update on this, the team has decided to use [minideb](https://github.com/bitnami/minideb) as the base image for this container instead of scratch. This stems from the limitations that a scratch image has for testing and to give us freedom to add future functionalities to the chart without having to work around scratch. It is true that it results in a bigger size for the container, but we think it is worthwhile. Do you have any strong opinions on this? 

I'll work on this at the start of next week and get back to you as soon as possible.

**Comment by mkilchhofer** [comment]
Hi @FraPazGal
This is okay for me. I already wondered where the benefit is, using an alternative container image (regarding the upstream one) which is also based "from scratch".
As soon as [Ephemeral Containers](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) will leave the alpha state, we can discuss this again. But for now I like containers which provides some basic tools for troubleshooting.

**Comment by mkilchhofer** [comment]
How is it going? Is there already progress on the new container image?
Is it required to have a the minideb variant of this container image to go on with the chart? :-)

**Comment by FraPazGal** [comment]
Hi @mkilchhofer, really sorry about the late response. The container is now ready and should be released to our CI/CD tools on Monday. I'll update you as soon as it is out!

**Comment by FraPazGal** [comment]
Hi @mkilchhofer,

I should have pinged you yesterday; the new image `bitnami/kubernetes-event-exporter:0-debian-10` is up. We have added `--help` as the default command, so you may need to change `args -> command` in `deployment.yaml`. Thank you for your patience!

**Comment by mkilchhofer** [comment]
Hi @FraPazGal ,
Glad to hear that :smiley: . I already updated this PR. Thanks for your work!
Would you review it?

**Review by FraPazGal — APPROVED** [review]

**Pull request merged**
